### ISSUE TITLE
Error while Compiling Assets: 

### ISSUE THREAD

**Issue by KrishnaChaitanya1991** [instructure/canvas-lms#1208]
### Steps to reproduce:
Hi Guys,
I am trying to setup canvas on my local system Ubuntu 14.04 but I am getting some errors while compiling assets:
Ruby - 2.4.1
Bundler - 1.14.6
Command:  rake canvas:compile_assets

Error Details: 
```
info fsevents@1.1.3: The platform "linux" is incompatible with this module.
info "fsevents@1.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[4/5] Linking dependencies...
warning " > instructure-ui@3.3.4" has unmet peer dependency "instructure-icons@^2.3.0 || >=3.0.0".
warning " > react-redux@4.4.5" has unmet peer dependency "redux@^2.0.0 || ^3.0.0".
warning "canvas-rce > react-transition-group@1.1.3" has incorrect peer dependency "react@^15.0.0".
warning "canvas-rce > react-transition-group@1.1.3" has incorrect peer dependency "react-dom@^15.0.0".
warning " > graphiql@0.11.5" has unmet peer dependency "prop-types@>=15.5.0".
warning " > graphiql@0.11.5" has incorrect peer dependency "react@^15.6.0".
warning " > graphiql@0.11.5" has incorrect peer dependency "react-dom@^15.6.0".
warning " > react-apollo@1.4.12" has unmet peer dependency "redux@^2.0.0 || ^3.0.0".
warning "tinymce-a11y-checker > react-aria-live@1.0.4" has incorrect peer dependency "react@15.x || ^16.0.0-rc".
[5/5] Building fresh packages...
[1/3] ⠠ node-sass: Downloading binary from https://github.com/sass/node-sass/releases/download/v4.7.2/linux-x64-57_binding.node
[2/3] ⠠ contextify:                                                                        ^
[-/3] ⠠ waiting...
[-/3] ⠠ waiting...
error /home/soumabha/work/canvas/node_modules/contextify: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /home/soumabha/work/canvas/node_modules/contextify
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.6.1
gyp info using node@8.0.0 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/soumabha/work/canvas/node_modules/contextify/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/home/soumabha/.node-gyp/8.0.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/home/soumabha/.node-gyp/8.0.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/soumabha/work/canvas/node_modules/contextify',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory `/home/soumabha/work/canvas/node_modules/contextify/build'
  CXX(target) Release/obj.target/contextify/src/contextify.o
../src/contextify.cc: In static member function ‘static v8::Local<v8::Context> ContextWrap::createV8Context(v8::Local<v8::Object>)’:
../src/contextify.cc:131:68: warning: ‘v8::Local<v8::Object> v8::Function::NewInstance() const’ is deprecated (declared at /home/soumabha/.node-gyp/8.0.0/include/node/v8.h:3674): Use maybe version [-Wdeprecated-declarations]
         Local<Object> wrapper = Nan::New(constructor)->NewInstance();
                                                                    ^
../src/contextify.cc:150:16: error: ‘class v8::ObjectTemplate’ has no member named ‘SetAccessCheckCallbacks’
         otmpl->SetAccessCheckCallbacks(GlobalPropertyNamedAccessCheck,
                ^
../src/contextify.cc: In static member function ‘static void ContextWrap::GlobalPropertyGetter(v8::Local<v8::String>, const Nan::PropertyCallbackInfo<v8::Value>&)’:
../src/contextify.cc:182:80: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated (declared at /home/soumabha/.node-gyp/8.0.0/include/node/v8.h:3227): Use maybe version [-Wdeprecated-declarations]
         Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
                                                                                ^
../src/contextify.cc: In static member function ‘static void ContextWrap::GlobalPropertyQuery(v8::Local<v8::String>, const Nan::PropertyCallbackInfo<v8::Integer>&)’:
../src/contextify.cc:209:67: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated (declared at /home/soumabha/.node-gyp/8.0.0/include/node/v8.h:3227): Use maybe version [-Wdeprecated-declarations]
         if (!Nan::New(ctx->sandbox)->GetRealNamedProperty(property).IsEmpty() ||
                                                                   ^
../src/contextify.cc:210:71: warning: ‘v8::Local<v8::Value> v8::Object::GetRealNamedProperty(v8::Local<v8::String>)’ is deprecated (declared at /home/soumabha/.node-gyp/8.0.0/include/node/v8.h:3227): Use maybe version [-Wdeprecated-declarations]
             !Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property).IsEmpty()) {
                                                                       ^
make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
make: Leaving directory `/home/soumabha/work/canvas/node_modules/contextify/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:285:23)
gyp ERR! stack     at emitTwo (events.js:125:13)
gyp ERR! stack     at ChildProcess.emit (events.js:213:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:197:12)
gyp ERR! System Linux 3.16.0-77-generic
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /home/soumabha/work/canvas/node_modules/contextify




rake aborted!
error running yarn install
/home/soumabha/work/canvas/lib/tasks/js.rake:58:in `block (2 levels) in <top (required)>'
/home/soumabha/work/canvas/lib/tasks/canvas.rake:31:in `block (3 levels) in <top (required)>'
/home/soumabha/work/canvas/lib/tasks/canvas.rake:7:in `log_time'
/home/soumabha/work/canvas/lib/tasks/canvas.rake:30:in `block (2 levels) in <top (required)>'
/usr/local/rvm/gems/ruby-2.4.1/gems/rake-12.2.1/exe/rake:27:in `<top (required)>'
/usr/local/rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `eval'
/usr/local/rvm/gems/ruby-2.4.1/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => js:yarn_install
(See full trace by running task with --trace)
```

**Comment by goindwalia** [comment]
+1

**Comment by xmavrck** [comment]
+1

**Comment by djbender** [comment]
Hi there!

Could you try again with the `master` branch? We think the problem was solved in fa9e6fb.
